src/main/java/Servlet/StudentServlet.java

```diff
@@ -0,0 +1,102 @@
+    private SessionFactory sessionFactory;
+
+    @Override
+    public void init() throws ServletException {
+        studentHibernate = new StudentHibernate();
+        sessionFactory = HibernateUtil.getSessionFactory();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        List<Student> students = studentHibernate.getAll();
+        req.setAttribute("studentsList", students);
+        req.getRequestDispatcher("/student.jsp").forward(req, resp);
+    }
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+            String action = request.getParameter("action");
+            if (action != null) {
+                if (action.equals("update")) {
+
+                    int id = Integer.parseInt(request.getParameter("id"));
+                    Student student = studentHibernate.getById(id);
+                    student.setName(request.getParameter("firstName"));
+                    student.setLastname(request.getParameter("lastName"));
+                    student.setAge(Integer.parseInt(request.getParameter("age")));
+
+                    Grooup grooup = new Grooup();
+                    grooup.setTitle(request.getParameter("grooupTitle"));
+                    student.setGrooup(grooup);
+
+                    City city = new City();
+                    city.setName(request.getParameter("cityName"));
+                    student.setCity(city);
+
+                    studentHibernate.update(student);
+
+                } else if (action.equals("delete")) {
+                    int id = Integer.parseInt(request.getParameter("id"));
+                    Student student = studentHibernate.getById(id);
+                    studentHibernate.delete(student);
+                }
+                response.sendRedirect(request.getContextPath() + "/student");
+                return;
+            }
+
+        Session session = sessionFactory.getCurrentSession();
+        session.beginTransaction();
+
+        try {
+            String firstName = request.getParameter("firstName");
+            String lastName = request.getParameter("lastName");
+            int age = Integer.parseInt(request.getParameter("age"));
+            String grooupTitle = request.getParameter("grooupTitle");
+            String cityName = request.getParameter("cityName");
+
+            Student student = new Student();
+            student.setName(firstName);
+            student.setLastname(lastName);
+            student.setAge(age);
+
+            Grooup grooup = new Grooup();
+            grooup.setTitle(grooupTitle);
+            session.saveOrUpdate(grooup);
+            student.setGrooup(grooup);
+
+            City city = new City();
+            city.setName(cityName);
+            session.saveOrUpdate(city);
+            student.setCity(city);
+
+            studentHibernate.save(student);
```

Comment:
It's strange a bit, when you using both session and `studentHibernate` which actually is like a DAO object.
You should use rith one or another for all database calls